packages/core/services/managers/map-layer-manager.ts

```diff
@@ -0,0 +1,82 @@
+import {Injectable, SimpleChanges} from '@angular/core';
+import {AgmTransitLayer} from '../../directives/transit-layer';
+import {AgmBicyclingLayer} from '../../directives/bicycling-layer';
+import {GoogleMapsAPIWrapper} from '../google-maps-api-wrapper';
+import {MapLayer, MapLayerOptions, GoogleMap} from '../google-maps-types';
+
+/**
+ * This class manages Transit and Bicycling Layers for a Google Map instance.
+ */
+
+@Injectable()
+export class MapLayerManager {
+    private _layers: Map<AgmTransitLayer|AgmBicyclingLayer, Promise<MapLayer>> =
+        new Map<AgmTransitLayer|AgmBicyclingLayer, Promise<MapLayer>>();
+
+    private static _layerOptions: string[] = [ 'visible'];
+
+    constructor(private _wrapper: GoogleMapsAPIWrapper) {}
+
+    /**
+     * Adds a map layer to a map instance. Keep track of that layer in map manager
+     * @param {AgmTransitLayer|AgmBicyclingLayer} layer - a MapLayer object
+     * @param {MapLayerOptions} options - MapLayerOptions options
+     * @returns void
+     */
+    addMapLayer(layer: AgmTransitLayer|AgmBicyclingLayer, options: MapLayerOptions): void {
+        const newLayer = this._wrapper.createMapLayer(options, layer.name());
+        this._layers.set(layer, newLayer);
+    }
+
+    /**
+     * Sets layer options
+     * @param {AgmTransitLayer|AgmBicyclingLayer} layer - a MapLayer object
+     * @param {SimpleChanges} changes related to the map later
+     * @returns Promise<void>
+     */
+    setOptions(layer: AgmTransitLayer|AgmBicyclingLayer, changes: SimpleChanges): Promise<void> {
+        const options = Object.keys(changes)
```

Comment:
SimpleChanges is part of angular component infrastructure, and should not be leaked to our Manager classes. I recommend going back to how it was originally.

Additionally, keeping in mind that there is only _ONE_ parameter for both bicycle layer, and transitlayer, it doesn't make sense do the entire `Object.keys`, `filter`, etc.

You can just do 
```typescript
if (changes['visible'] != null) {
  return this.layerManager.setVisible(this, changes['visible'].currentValue);
}
```
Of course keep the above lines of code in the directive's `ngOnChanges` method, and not in manager.